tempodb/encoding/vparquet2/block_traceql.go

```diff
@@ -66,6 +94,9 @@ func (s *span) attributesMatched() int {
 	if len(s.id) > 0 {
 		count++
 	}
+	if s.nestedSetLeft > 0 || s.nestedSetRight > 0 || s.nestedSetParent > 0 {
+		count++
```

Comment:
should these be evaluated independently and each increment count separately? since they come in as different conditions?

Reply:
oh, reading below. it's kind of impossible to exactly count the number of attributes. just incrementing once is playing it safe?

Reply:
Yeah each operation uses a different subset. We could know the case for a single operator, but it's not possible to distinguish between child (which loads left + parent), and parent (which loads only parent) for example.  However it didn't seem necessary to increment exactly, just by 1 is enough to ensure the span passes through the filter logic in spanCollector.

Reply:
Want to add more detail on why it works to only increment by 1.  It's because of this line:  https://github.com/grafana/tempo/blob/main/pkg/traceql/ast.go#L238   Any spanset operation unchecks the AllConditions optimization, which means a span only has to meet 1 condition to be passed up to the engine.  The exact count is only needed when the optimization is present (only spans meeting all are passed up to the engine).